src/psyclone/tests/psyir/backend/fortran_gen_decls_test.py

```diff
@@ -253,14 +253,17 @@ def test_gen_decls_routine_unresolved(fortran_writer):
 
 def test_gen_decls_routine(fortran_writer):
     '''Test that the gen_decls method raises an exception if the interface
-    of a routine symbol is not an UnresolvedInterface or
-    ImportInterface, unless there's a wildcard import from a
-    Container.
+    of a routine symbol is not an ImportInterface, unless there's a wildcard
+    import from a Container.
 
     '''
-    # Check a user-defined routine symbol with an (unsupported)
-    # ArgumentInterface
     symbol_table = SymbolTable()
+    # Check that a RoutineSymbol representing an intrinsic is OK
+    symbol_table.add(RoutineSymbol("nint", interface=UnresolvedInterface()))
```

Comment:
This check passes because a nint is not yet an IntrinsicSymbol (used by an IntrinsicCall), but ultimately if we had a RoutineSymbol("nint") it would be something redeclaring the intrinsic (which is valid Fortran) and would need the declaration or import statement for it, right?
If this is the case we may want a TODO here (or better in the code that lets this happen). Do we have an issue to port all intrinsics to proper IntrinsicSymbols?

Reply:
Actually, no. The comment was misleading - this is nothing to do with intrinsics. Since Fortran permits the resolution of symbols to be deferred until link time then all is fine as long as the routine is made available in some other file scope. (@rupertford changed the PSyIR backend to allow for this situation a couple of months ago.) I've therefore updated the comment but left the code unchanged.

Reply:
In fact, the first part of this test is already covered by the previous one so I've updated this test to purely check for the exception that we raise if the interface to the RoutineSymbol is something unexpected.